### PR TITLE
ci: add a memory sanitiser test run

### PR DESCRIPTION
**Pull request by paulidale**
This omission noted in #15950 which includes the RSA test failure fix.  The RSA test failure is thus expected here.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/afalgtest.c [hunk 1]
```diff
@@ -48,27 +48,39 @@ static int test_afalg_aes_cbc(int keysize_idx)
...
+        "\xe3\x53\x77\x9c\x10\x79\xae\xb8\x27\x08\x94\x2d\xbe\x77\x18\x1a"
+        "\x2d";
+    static const unsigned char encresult_192[BUFFER_SIZE] =
+        "\xf7\xe4\x26\xd1\xd5\x4f\x8f\x39\xb1\x9e\xe0\xdf\x61\xb9\xc2\x55"
+        "\xeb";
+    static const unsigned char encresult_256[BUFFER_SIZE] =
+        "\xa0\x76\x85\xfd\xc1\x65\x71\x9d\xc7\xe9\x13\x6e\xae\x55\x49\xb4"
+        "\x13";
+
+#ifdef OSSL_SANITIZE_MEMORY
```
Comment (paulidale) [inline test/afalgtest.c:75]: Note to reviewers: this stanza is the only real change here.  The rest is reformatting and rearrangement.

**Comment by paulidale** [comment]
As noted, the CI failure is relevant but is being dealt with in another PR.

**Review by slontis — APPROVED** [review]
Should only be merged after #15950 has been merged.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
#15950 has been merged, rebased this to include it and squashed.  The CIs will reveal any problems.  There are no code changes.

**Comment by paulidale** [comment]
Merged, thanks for the reviews.

**Pull request closed without merging**